pkg/controllers/nodeclaim/termination/suite_test.go

```diff
@@ -216,4 +217,31 @@ var _ = Describe("Termination", func() {
 			ExpectExists(ctx, env.Client, node)
 		}
 	})
+	It("should retry nodeClaim deletion when cloudProvider returns retryable error", func() {
+		ExpectApplied(ctx, env.Client, nodePool, nodeClaim)
+		ExpectReconcileSucceeded(ctx, nodeClaimLifecycleController, client.ObjectKeyFromObject(nodeClaim))
+
+		nodeClaim = ExpectExists(ctx, env.Client, nodeClaim)
+		_, err := cloudProvider.Get(ctx, nodeClaim.Status.ProviderID)
+		Expect(err).ToNot(HaveOccurred())
+
+		node := test.NodeClaimLinkedNode(nodeClaim)
+		ExpectApplied(ctx, env.Client, node)
+
+		// Expect the node and the nodeClaim to both be gone
+		Expect(env.Client.Delete(ctx, nodeClaim)).To(Succeed())
+		ExpectReconcileSucceeded(ctx, nodeClaimTerminationController, client.ObjectKeyFromObject(nodeClaim)) // triggers the node deletion
+		ExpectFinalizersRemoved(ctx, env.Client, node)
+		ExpectNotFound(ctx, env.Client, node)
+
+		cloudProvider.NextDeleteErr = cloudprovider.NewRetryableError(fmt.Errorf("underlying instance not terminated"))
+		err = cloudProvider.Delete(ctx, nodeClaim)
+		Expect(cloudprovider.IsRetryableError(err)).To(BeTrue())
+		ExpectReconcileSucceeded(ctx, nodeClaimTerminationController, client.ObjectKeyFromObject(nodeClaim))
```

Comment:
I think if the reconcile succeeds, we get back the result. Can we check that result here to see that we are doing the back-off that we expect? e.g. 10s?